src/ethereum/api/Bridge.ts

```diff
@@ -1,65 +1,100 @@
-import { CollectionAttributes, CollectionWithItems } from '../../Collection'
-import { ItemAttributes, CollectionItem } from '../../Item'
-import { CollectionFields, CollectionFragment, ItemFragment } from './fragments'
+import { CollectionAttributes, Collection } from '../../Collection'
+import { ItemAttributes } from '../../Item'
+import { ItemFragment, CollectionFragment } from './fragments'
+import { collectionAPI } from './collection'
 
 export class Bridge {
-  consolidateCollections(
-    dbCollections: CollectionWithItems[],
-    remoteCollections: Partial<CollectionAttributes>[],
-    remoteItems: Partial<CollectionItem>[] = []
+  static consolidateCollections(
+    dbCollections: CollectionAttributes[],
+    remoteCollections: CollectionFragment[]
   ) {
-    const collections: CollectionWithItems[] = []
-    const remoteAddresses = remoteCollections.map(
-      collection => collection.contract_address
-    )
+    const collections: CollectionAttributes[] = []
 
     for (const dbCollection of dbCollections) {
-      const index = remoteAddresses.indexOf(dbCollection.contract_address)
-      const remoteCollection = remoteCollections[index]
-      const collection =
-        index === -1 ? dbCollection : { ...dbCollection, ...remoteCollection }
-
-      const remoteCollectionItems = remoteItems.filter(
-        item => item.collection_id === collection.id
-      )
-      collection.items = this.consolidateItems(
-        dbCollection.items,
-        remoteCollectionItems
+      const remoteCollection = remoteCollections.find(
+        remoteCollection =>
+          remoteCollection.id === dbCollection.contract_address
       )
+      const collection = remoteCollection
+        ? Bridge.mergeCollection(dbCollection, remoteCollection)
+        : dbCollection
 
       collections.push(collection)
     }
 
     return collections
   }
 
-  consolidateItems(
-    dbItems: CollectionItem[],
-    remoteItems: Partial<CollectionItem>[]
+  static async consolidateItems(
+    dbItems: ItemAttributes[],
+    remoteItems: ItemFragment[]
   ) {
     const items: ItemAttributes[] = []
 
+    // Get collections from DB
+    const collectionIds = dbItems.reduce<string[]>((ids, dbItem) => {
+      if (dbItem.collection_id) {
+        ids.push(dbItem.collection_id)
+      }
+      return ids
+    }, [])
+
+    const dbResults = await Promise.all(
+      collectionIds.map(collectionId =>
+        Collection.findOne<CollectionAttributes>(collectionId)
+      )
+    )
+    const dbCollections = dbResults.reduce((obj, result) => {
+      if (result) {
+        obj[result.id] = result
+      }
+      return obj
+    }, {} as Record<string, CollectionAttributes>)
+
+    // Get remote collections
+    const contractAddresses = Object.values(dbCollections).map(
```

Comment:
You could just use dbResults here right?
```suggestion
    const contractAddresses = dbResults.map(
```

Reply:
`dbResults` is `(Collection | undefined)[]` that's why I didn't used that, the `undefined` values are filtered by the `if (result)` when I convert the array into object